### PR TITLE
Remove unused dependencies 2023/09 edition

### PR DESCRIPTION
**Pull request by citizen-stig**
# Description

cargo udeps + cargo machete.

Only removed those which I felt comfortable removing. If something breaks, this is good opportunity to add tests

### CODE REVIEW

**Review by neysofu — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -21,16 +21,18 @@ install-dev-tools:  ## Installs all necessary cargo helpers
...
 	cargo install cargo-nextest --locked
+	cargo install cargo-risczero
+	cargo risczero install
 
 lint:  ## cargo check and clippy. Skip clippy on guest code since it's not supported by risc0
 	## fmt first, because it's the cheapest
-	cargo fmt --all --check
-	cargo check --all-targets --all-features
+	cargo +nightly fmt --all --check
+	cargo +nightly check --all-targets --all-features
```
Comment (neysofu) [inline Makefile]: We only want `+nightly` for `cargo fmt`, all the other commands need to use stable.

**Review by neysofu — CHANGES_REQUESTED** [review]
(Just to make sure the PR doesn't get auto-merged with the `+nightly` usage)

File: Makefile (hunk 1 above)
Comment (citizen-stig) [reply Makefile]: Done!

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -57,7 +57,7 @@ env:
 # Source: <https://stackoverflow.com/a/72408109/5148606>
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
-  # Except in `nightly` and `stable` branches! Any cancelled job will cause the
+  # Except in `nightly` and `stable` branches! Any canceled job will cause the
```
Comment (citizen-stig) [inline .github/workflows/rust.yml]:
```suggestion
  # Except in `nightly` and `stable` branches! Any cancelled job will cause the
```

**Review by neysofu — APPROVED** [review]

**Pull request merged**